ExtractUserDataActivity/handler.ts

```diff
@@ -463,27 +469,49 @@ export const queryAllUserData = (
       })
     ),
     // step 3: queries notifications statuses
-    TE.chain(
+    TE.bind("notificationStatuses", ({ notifications }) =>
+      findAllNotificationStatuses(notificationStatusModel, notifications)
+    ),
+    // step 4: queries profile service preferences
+    TE.bind("servicesPreferences", () =>
+      pipe(
+        servicePreferencesModel.findAllByFiscalCode(fiscalCode),
+        flattenAsyncIterator,
+        asyncIteratorToArray,
+        promise =>
```

Comment:
Can't we just use `fromPromiseEither` or define something similar?

Reply:
Isn't clear to me what do you mean with `fromPromiseEither`. In here we have a Promise<ReadonlyArray<Either>> and the `Either`s are not relevant in the TE transformation. IMHO the `tryCatch` is the best method to transform a Promise to a TE where the left value is obtained only on the promise failure.

Reply:
i created another function because the promise mentioned above was returning an `either[]`. done in https://github.com/pagopa/io-functions-admin/pull/191/commits/14030001890cd960f5309d3f6bac471151235df0

Reply:
I think that with a separated function the code is less readable in this case, but i leave on you @michaeldisaro @arcogabbo

Reply:
On new functions pls add a detailed JS doc :D